### PR TITLE
Fix selection menu not showing after clear

### LINKED ISSUE #35998: Selection menu doesn't show after entering text then clearing/deleting
After https://github.com/flutter/flutter/issues/33293 was merged, a [commenter pointed out](https://github.com/flutter/flutter/issues/33293#issuecomment-510500688) that this similar problem still exists.

### Steps to reproduce

  1. Run an app with a TextField (see below for an example).
  1. Long press the field and the selection menu should show (correct).
  1. Type some text and long press outside of the text to show the menu again (correct).
  1. Delete the text so the field is empty and long press again.  The menu won't be shown (incorrect!).

### Code

```dart
import 'package:flutter/material.dart';\

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(),
      title: 'Text Field Test',
      home: MyApp(),
    );
  }
}
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("TextField Test"),
      ),
      body: Center(
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          mainAxisAlignment: MainAxisAlignment.center,
          child: TextField(),
        ),
      ),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by justinmc**
## Description

If you long press an empty text field, then the selection menu shows, but if you type some content and then delete it all so it's empty again, the menu doesn't show.  This PR fixes that.

## Related Issues

Closes https://github.com/flutter/flutter/issues/35998

## Tests

Tested adding text, deleting it, and then showing the menu in editable_text_test.dart.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -1360,11 +1360,6 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
-  void _hideSelectionOverlayIfNeeded() {
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/editable_text.dart]: I think some of the problem came from the fact that there are two subtly different methods with confusing names: `_hideSelectionOverlayIfNeeded` and `hideToolbar`.  I got rid of this one altogether and I think it makes more sense now.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 2]
```diff
@@ -29,6 +29,11 @@ enum HandlePositionInViewport {
 void main() {
   setUp(() {
     debugResetSemanticsIdCounter();
+    controller = TextEditingController();
+  });
+
+  tearDown(() {
+    controller = null;
```
Comment (HansMuller) [inline packages/flutter/test/widgets/editable_text_test.dart]: `controller.dispose()` first?

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 3]
```diff
@@ -531,6 +536,21 @@ void main() {
     expect(state.showToolbar(), true);
     await tester.pump();
     expect(find.text('PASTE'), findsOneWidget);
+
+    // Hide the toolbar and clear the text and selection.
```
Comment (HansMuller) [inline packages/flutter/test/widgets/editable_text_test.dart]: Maybe factor this out as a separate test and mark it as a regression test for https://github.com/flutter/flutter/issues/35998

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 2 above)
Comment (justinmc) [reply packages/flutter/test/widgets/editable_text_test.dart]: Good catch.

**Pull request merged**